test/channels.jl

```diff
@@ -578,3 +579,39 @@ let c = Channel(3)
     close(c)
     @test repr(MIME("text/plain"), c) == "Channel{Any}(3) (closed)"
 end
+
+# PR #41833
+@testset "n_avail(::Channel)" begin
+    # Buffered: n_avail() = buffer length + number of waiting tasks
+    let c = Channel(2)
+        @test n_avail(c) == 0;   put!(c, 0)
+        @test n_avail(c) == 1;   put!(c, 0)
+        @test n_avail(c) == 2;   yield(@task put!(c, 0))
+        @test n_avail(c) == 3;   yield(@task put!(c, 0))
+        @test n_avail(c) == 4
+        # Test n_avail(c) after interrupting a task waiting on the channel
+                                t = @task put!(c, 0)
+                                yield(t)
+        @test n_avail(c) == 5
+                                @async Base.throwto(t, ErrorException("Exit put!"))
+                                try wait(t) catch end
+        @test n_avail(c) == 4
```

Comment:
Is this the test for `did_buffer === false` case?

Reply:
Yes.